src/cache/cache_las.c

```diff
@@ -212,12 +252,30 @@ __wt_las_cursor(
 	if (F_ISSET(session, WT_SESSION_LOOKASIDE_CURSOR))
 		*cursorp = session->las_cursor;
 	else {
-		__wt_spin_lock(session, &conn->las_lock);
-		*cursorp = conn->las_session->las_cursor;
+		for (;;) {
+			__wt_spin_lock(session, &cache->las_lock);
+			for (i = 0; i < WT_LAS_NUM_SESSIONS; i++) {
+				if (!cache->las_session_inuse[i]) {
+					*cursorp =
+					    cache->las_session[i]->las_cursor;
+					cache->las_session_inuse[i] = true;
+					break;
+				}
+			}
+			__wt_spin_unlock(session, &cache->las_lock);
+			if (*cursorp != NULL)
+				break;
+			/*
+			 * If all the lookaside sessions are busy, stall.
+			 *
+			 * XXX better as a condition variable.
+			 */
+			__wt_sleep(0, 1000);
```

Comment:
I haven't seen threads waiting here in my testing so far, but it's workload-dependent and fixed sleeps are always bad for some case.  Maybe a followup ticket?